legend-engine-xts-sql/legend-engine-xt-sql-postgres-server/src/main/java/org/finos/legend/engine/postgres/config/Builder.java

```diff
@@ -42,7 +45,14 @@ public static SessionsFactory buildSessionFactory(ServerConfig serverConfig)
         else if (serverConfig.getHandler().getType() == HandlerType.LEGEND)
         {
             LegendHandlerConfig config = (LegendHandlerConfig) serverConfig.getHandler();
-            LegendTdsClient client = new LegendTdsClient(config.getProtocol(), config.getHost(), config.getPort());
+            LegendExecutionService client = new LegendExecutionService(new LegendHttpClient(config.getProtocol(), config.getHost(), config.getPort()));
+            return new LegendSessionFactory(client);
+        }
+        else if (serverConfig.getHandler().getType() == HandlerType.STATIC)
```

Comment:
This builder with a lot of `if` conditions doesn't seem to actually build anything. Can we move all the logic into the `ServerConfig` and the `HandlerConfig` and delete this builder?

Reply:
done